corehq/sql_db/util.py

```diff
@@ -81,24 +82,34 @@ def paginate_query_across_partitioned_databases(model_class, q_expression, annot
     :param q_expression: An instance of django.db.models.Q representing the
     filter to apply
 
-    :param annotate: (optional) If specified, should by a dictionary of annotated fields
+    :param annotate: (optional) If specified, should be a dictionary of annotated fields
     and their calculations. The dictionary will be splatted into the `.annotate` function
 
+    :param values: (optional) If specified, should be a list of values to retrieve rather
+    than retrieving entire objects. If `pk` is not defined in `values`, then the values
+    returned will be a tuple of (pk + *values)
+
     :return: A generator with the results
     """
     db_names = get_db_aliases_for_partitioned_query()
+    sort_col = 'pk'
+
+    return_values = None
+    if values and sort_col not in values:
+        return_values = ['pk'] + values
 
     for db_name in db_names:
         qs = model_class.objects.using(db_name)
         if annotate:
             qs = qs.annotate(**annotate)
 
         qs = qs.filter(q_expression)
-        sort_col = 'pk'
         value = 0
         last_value = qs.order_by('-{}'.format(sort_col)).values_list(sort_col, flat=True).first()
         if last_value is not None:
             qs = qs.order_by(sort_col)
+            if return_values:
+                qs.values_list(*return_values)
```

Comment:
This should be
```suggestion
                qs.values_list(named=True, *return_values)
```
so you can use named field values and so

https://github.com/dimagi/commcare-hq/blob/22f4d86fb6d4fde16ed2b5abfb00b3755389328d/corehq/sql_db/util.py#L116

continues to work.

Reply:
Thanks for catching this. Unfortunately `named` was introduced in django 2.0 (and I almost punted this change because not using namedtuples was annoying), so your change will need to wait until after python 3

Reply:
We could implement named tuples pretty easily here ourselves. Not sure if it's worth it though?